gammapy/datasets/map.py

```diff
@@ -2510,13 +2509,31 @@ def update(self, exposure, psf, edisp, geom):
         # TODO: simplify and clean up
         log.debug("Updating model evaluator")
         # cache current position of the model component
+        geom2d = geom.to_image()
+        if (
+            self.model.position is not None
+            and isinstance(geom2d, WcsGeom)
+            and not geom2d.contains(self.model.position)[0]
+        ):
+            coords = SkyCoord(
+                geom2d.get_coord()["lon"].squeeze(),
+                geom2d.get_coord()["lat"].squeeze(),
+                frame=geom2d.frame,
+            )
+            separation = coords.separation(self.model.position)
+            separation[exposure.sum_over_axes().data.squeeze() == 0] = np.inf
+            ilon, ilat = np.where(separation == np.min(separation))
+            irf_position = coords[ilon[0], ilat[0]]
+            log.warning(
+                f"Center position for {self.model.name} model is outside dataset geom, using nearest IRF defined within geom"
+            )
+        else:
+            irf_position = self.model.position
```

Comment:
Maybe also introduce `MapEvaluator.irf_position`? Might be useful for debugging in some cases...